demos/eret_hvc_smc_wrapper.cc

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <array>
```

Comment:
Somewhere in this commit we need to have an end-to-end description of what's going on. Maybe that goes in the commit message, but it's probably more useful in the code -- here, as a comment, or in a Markdown file alongside. But we need to make special effort to make these demos useful to an audience that isn't intimately familiar with the vulnerabilities involved. This one in particular is really tricky, and we need to explain the mechanics involved, what these instructions do, what behavior we're showing instead, why we need a kernel module to do that, what the module is for, etc.

Reply:
Done.